sklearn/linear_model/tests/test_sparse_coordinate_descent.py

```diff
@@ -137,17 +142,20 @@ def make_sparse_data(
 
     # generate training ground truth labels
     y = np.dot(X, w)
-    X = sp.csc_matrix(X)
+    X = csc_container(X)
     if n_targets == 1:
         y = np.ravel(y)
     return X, y
 
 
-def _test_sparse_enet_not_as_toy_dataset(alpha, fit_intercept, positive):
+@pytest.mark.parametrize("csc_container", CSC_CONTAINERS)
+def _test_sparse_enet_not_as_toy_dataset(alpha, fit_intercept, positive, csc_container):
```

Comment:
I don't think the parametrization should be on a protected function. I think this function should take as an argument the csc_container but the parametrization should happen on the test below which actually uses this.